### PR TITLE
feat: integrate EMA Oracle

### PR DESCRIPTION
**Pull request by apopiak**
Integrate the EMA oracle from https://github.com/galacticcouncil/warehouse/pull/76 into the Basilisk runtime.

## Description

This PR adds the EMA oracle pallet and configures the runtime to ingest xyk trades and liquidity changes into it.
Supported periods are configured to `LastBlock`, `Hour`, `Day` and `Week`.

The oracle values are not used, yet, to allow for observation.

supersedes #557 

## Related Issue

https://app.clickup.com/t/2v0w820

## How Has This Been Tested?

Includes an integration tests, most of the tests are in the pallet and its corresponding math package.

## Checklist:

- [x] I have updated the documentation if necessary.
- [x] I have added tests to cover my changes, regression test if fixing an issue.
- [x] This is a breaking change.

### CODE REVIEW

File: runtime/basilisk/src/lib.rs [hunk 1]
```diff
@@ -931,6 +932,23 @@ impl pallet_route_executor::Config for Runtime {
 	type WeightInfo = weights::route_executor::BasiliskWeight<Runtime>;
 }
 
+// constants need to be in scope to be used in generics
+use pallet_ema_oracle::{MAX_PERIODS, MAX_TRADES};
+
+parameter_types! {
+	pub SupportedPeriods: BoundedVec<OraclePeriod, ConstU32<MAX_PERIODS>> = BoundedVec::truncate_from(
+		vec![OraclePeriod::LastBlock, OraclePeriod::TenMinutes, OraclePeriod::Day, OraclePeriod::Week]
```
Comment (apopiak) [inline runtime/basilisk/src/lib.rs:940]: @mrq1911 @jak-pan We will need to decide on what this value should be.

**Review by enthusiastmartin — APPROVED** [review]
lgtm

**Comment by apopiak** [comment]
This should only need a rebenchmark based on the new max unique entries (now 30).

**Comment by apopiak** [comment]
Note: needs update to https://github.com/galacticcouncil/warehouse/pull/188

File: runtime/basilisk/src/lib.rs [hunk 2]
```diff
@@ -494,7 +495,7 @@ impl pallet_xyk::Config for Runtime {
 	type MaxInRatio = MaxInRatio;
 	type MaxOutRatio = MaxOutRatio;
 	type CanCreatePool = pallet_lbp::DisallowWhenLBPPoolRunning<Runtime>;
-	type AMMHandler = ();
+	type AMMHandler = pallet_ema_oracle::OnActivityHandler<Runtime>;
```
Comment (mrq1911) [inline runtime/basilisk/src/lib.rs:498]: shouldn't be xyk reweighted now with oracle?

Comment (apopiak) [reply runtime/basilisk/src/lib.rs:498]: yeah

**Pull request merged**